### DISCUSSION TITLE
Restrict Guest users possibility to create new stores.

### DISCUSSION THREAD

**Discussion by Jossec101** [btcpayserver/btcpayserver#3618] [Ideas / Feature Requests]
Hi there!

I initially thought to report this as a security issue but perhaps it was intended, we have realized that guest users can create new stores, is this something intended? In our honest opinion, the creation of stores should be only managed by privileged users like server admins.

Best!

**Comment by ishristov** [comment]
Well the server admin is only 1 (afaik) and sometimes you operate a server as host where each new user should be able to create a store for themselves. But I agree that sometimes you'd like no one to create new stores (i.e. when you want to manage that for them).

Perhaps it would make sense to add a new server option (for the admin) that says "Allow users to create new stores". It doesn't make sense to limit that on the guest/owner level as the guest/owner is a store-specific privilege, not a global setting for the user. At least that's how I understand it.

**Comment by Jossec101** [comment]
@ishristov Yes an option to disable guest users to create stores would solve this definitely, good idea. ;)

**Reply by ishristov** [reply]
I was talking about "non-admin" users, not about guest/owner users (which from what I understand is only a store specific role).

**Comment by bolatovumar** [comment]
@Jossec101 @ishristov if guest user can't create a store then they can't really do anything in BTCPay. This is what it looks like when a guest user first signs up. As you can see there isn't anything you can do here if you don't have a store.

![Capture](https://user-images.githubusercontent.com/1934678/167996507-99c54e8d-ce26-4f6f-9695-b14a400fbd5b.PNG)

**Reply by Jossec101** [reply]
True but that does not mean that having a "disable guest users from creating stores" feature might be a bad a idea for certain use cases..

**Reply by pavlenex** [reply]
I assume the use case here is:
- Server admin creates a store, assigns it to a a guest account, but wants to limit that account from creating new stores, though it shouldn't matter that much as those stores are created under their own account.
I think we can flash this out once we implemented proper "store manager" role that we've been discussing lately.

**Reply by Jossec101** [reply]
> I assume the use case here is:
> 
> * Server admin creates a store, assigns it to a a guest account, but wants to limit that account from creating new stores, though it shouldn't matter that much as those stores are created under their own account.
>   I think we can flash this out once we implemented proper "store manager" role that we've been discussing lately.

Exactly, we want guest users to have a view of the invoices and pull payments UI of certain pre-assigned stores  but not allow them to create new stores.

**Reply by ishristov** [reply]
Not sure about @Jossec101's use case, but mine with regard to this is "I am the server admin. Only I should create stores. I add some guest and some owners to some stores that I've created for them, but neither of them should be able to create new stores, regardless of their role. Even if some user has an 'owner' role, I still don't want them to create stores. That's my responsibility only".

**Reply by pavlenex** [reply]
Would we want to limit them to creating stores entirely, or create any stores, after the first one, meaning only one store?

**Reply by Jossec101** [reply]
In our case entirely, we wouldn't like for guests to create new stores.

**Reply by Kukks** [reply]
> * I think we can flash this out once we implemented proper "store manager" role that we've been discussing lately.

No, Store manage roller is a store user role(current options are `store owner` and `store guest` ). This feature request is for a user role (current options are `server admin` or none) .

**Reply by ishristov** [reply]
> In our case entirely, we wouldn't like for guests to create new stores.

And in ours, we wouldn't any user that is not a server admin to create a new store.
Just like there is a server setting `Disable new user registration on the server`, there can be `Disable non-server admins to create new stores`.
But to match @Jossec101's use case, this privilege (to create new stores) might be added/edited individually per user. One default setting `Disable non-server admins to create new stores by default` but then allow each user to be edited separately in the Users sections, despite their guest/owner role.

**Comment by BoGitter** [comment]
+1 for store manager role (but not allowed to create store)
